lib/endpoint/endpoint.go

```diff
@@ -0,0 +1,21 @@
+package endpoint
+
+import (
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Endpoint knows how to connect with a Transport or a Transfer
+type Endpoint interface {
+	// NamespacedName returns a ns name to identify this endpoint
+	NamespacedName() types.NamespacedName
+	// Hostname returns a hostname for the endpoint
+	Hostname() string
+	// BackendPort returns a port which can be used by the application behind this endpoint
+	// If an endpoint does not support a choice of backend port it will return error
```

Comment:
Where is the error return for the function if this sentence is correct?

Reply:
the comment is inaccurate, part of a bad refactor. will correct it.